gateway/api/ray.py

```diff
@@ -157,7 +157,10 @@ def submit_job(job: Job) -> Job:
 
 
 def create_ray_cluster(
-    user: Any, cluster_name: Optional[str] = None, cluster_data: Optional[str] = None
+    user: Any,
+    cluster_name: Optional[str] = None,
+    cluster_data: Optional[str] = None,
+    job_config: Optional[JobConfig] = None,
```

Comment:
we accept optional job_config here, but later on we always act like it is not None.

Here we should check for None and fall back to default from settings.